kinto_client/__init__.py

```diff
@@ -117,265 +110,113 @@ def request(self, method, url, data=None, permissions=None, **kwargs):
-        self.name = utils.slugify(name)
-        self.uri = "%s/collections/%s" % (self.bucket.uri, self.name)
-        self.permissions = Permissions('collection')
-        self.data = None
-
-        if load:
-            # XXX put this logic in a separate method.
-            bucket_name = self.bucket.name
-            method = 'put' if create and bucket_name != 'default' else 'get'
-            request_kwargs = {}
-            if method == 'put' and permissions is not None:
-                request_kwargs['permissions'] = permissions
-
-            info, _ = self.session.request(method, self.uri, **request_kwargs)
-            self.data = info['data']
-            self.permissions = Permissions('collection', info['permissions'])
-
-    def _get_record_uri(self, record_id):
-        return '%s/records/%s' % (self.uri, record_id)
-
-    def get_records(self):
+    def get_records(self, collection=None, bucket=None):
         """Returns all the records"""
         # XXX Add filter and sorting.
-        records_uri = '%s/records' % self.uri
-        resp, _ = self.session.request('get', records_uri)
-
-        # XXX Support permissions for GET /records.
-        return [Record(data=data, collection=self, session=self.session)
-                for data in resp['data']]
-
-    def get_record(self, id):
-        return Record(id=id, collection=self, session=self.session)
-
-    def save_record(self, record):
-        # XXX Chose between different approaches.
-        return record.save()
-
-    def save_records(self, records):
-        # XXX Enhance this with a batch request.
-        for record in records:
-            record.save()
-
-    def create_record(self, data, permissions=None):
-        return Record(data=data, permissions=permissions, collection=self,
-                      create=True, session=self.session)
-
-    def delete_record(self, id):
-        record_uri = self._get_record_uri(id)
-        resp, _ = self.session.request('delete', record_uri)
-
-    def delete_records(self, records):
-        # XXX Enhance this with a batch request.
-        # with self.session.batch() as batch:
-        #     for record in records:
-        #         batch.request('delete', self._get_record_uri(record.id))
-        for record in records:
-            self.delete_record(record.id)
-
-    def delete(self):
-        resp, _ = self.session.request('delete', self.uri)
+        # XXX Add support for pagination.
+        endpoint = self._get_endpoint('records', bucket, collection)
+        resp, _ = self.session.request('get', endpoint)
         return resp['data']
 
+    def get_record(self, id, collection=None, bucket=None):
+        endpoint = self._get_endpoint('record', bucket, collection, id)
+        resp, _ = self.session.request('get', endpoint)
+        return resp
+
+    def create_record(self, data, id=None, collection=None, permissions=None,
+                      bucket=None):
+        id = id or data.get('id', None) or str(uuid.uuid4())
```

Comment:
In the mean time, I would make a POST rather than a PUT on an empty id to let the server create the id.